### DISCUSSION TITLE
HDHomerun network quirk noted with SageTV.

### DISCUSSION THREAD

**Discussion by MABarta** [OpenSageTV/Community#122] [Capture Devices]
And so far as I could tell, the behavior applied exclusively to SageTV, even open DCT could see/interact with them from the same computer. PlexDVR could also access the tuners from a similarly configured server.

Setup: SageTV Server on a static IP, we'll say 192.168.3.40, with a subnet of 255.255.255.0 (plex server was a close neighbor numerically)
3 HD Homerun tuners on a dynamic IP assignment from the router because Silicon Dust wants us to feel pain.
Wireless Router gets replaced, it's new DHCP server is setup to use 192.168.3.100+ assignments for non-static devices, thanks to a murky configurator, I didn't check the subnet assignment, turns out it was 255.255.253.0 no idea why a consumer device would need such a large subnet, but I guess it might be for some VPN activities.

Result, SageTV breaks.  SageTV will no longer talk to the HD Homerun units, although OpenDCT can still detect them, however, SageTV also decides to disallow using OpenDCT as an intermediary. PlexDVR happily works, and the Silicondust app works on the smart TV's(of course, using DHCP, their subnets likely matched). After a fair bit of running around in circles, I looked at the DHCP assignments for the HDHomeRun units and finally noticed the subnet assignment was different. Checking the numbers, no reason why they shouldn't be able to talk to each other, but went into the router and fixed that oversight. Some time later, the HDHomeRun unit updates its subnet, and magically, SageTV begins talking to them again.

It wasn't a firewall problem, in the typical sense at least. Not sure why SageTV was the only one who decided not to play on that one. But in the event someone else encounters something like this in the future, it's documented. Or maybe someone can dive into the code and figure out exactly what was going on there, assuming they can reproduce it.

**Comment by wayner9** [comment]
Weird.  Just to clarify - Is your theory that not having a subnet of 255.255.255.0 will cause problems with SageTV?

**Reply by MABarta** [reply]
Theory is non-matching sub-nets will break SageTV, at least with a HDHomeRun tuner. Even if the HDHomeRun subnet overlaps with SageTV.

Didn't test for changing the server to 255.255.253.0 to see if that would have fixed things, would have been faster/easier in hindsight.

**Reply by wayner9** [reply]
I believe I have heard of equipment in the past that seemed to always uses a /24 subnet mask regardless of what you entered.

**Comment by trk2** [comment]
I don't understand why you are not assigning the HD Homerun devices a fixed IP?  I have noticed there can be an order of operations issue where Sage (assuming running as a service) can start before OpenDCT has finished its search for devices and therefore the devices will not be available to Sage unless you stop and restart the service.

**Reply by eelpout** [reply]
yeah, I have had all my tuner IP's assigned by MAC address by the router for years and haven't had any issues. This includes DirecTV devices and an HDHomeRun.

**Reply by MABarta** [reply]
At least for the HD Homerun tuners I have, there is no option except dynamic IP assignment on the tuner end. So any IP assignments would need to happen on the side of the DHCP Server. I normally address that by reserving the IP assignments after the first run of a new router--which takes care of things until a new router is introduced, as happened here. Such upgrades/replacements are rare and infrequent happening once every few years, so not considered much of an issue on my part, and it hasn't been previously.

Normally it creates a few minutes of extra work on SageTV, but is generally faster and more productive than seeking out the DHCP server settings on the newest consumer router to see if static assignments are even possible. Except this time the new router threw the curveball of changing the subnet mask to something more permissive, and hiding it in a different part of the UI. So even if I had changed the IP address back to what it was before, because the subnet had changed relative to the SageTV server, it wouldn't work.
